ReactTask/src/components/CreateAppointment/CreateAppointment.js

```diff
@@ -44,27 +73,30 @@ const CreateAppointment = ({ switchContent }) => {
         <StylItemChooseDoctor contentText='Select an available timeslot'>
           <StylBoxColumnFlexMargin dataReady={isFilledData}>
             <StylBoxChooseTime>
-              {possibleTimeArr.map((item, index) => (
-                <StylBtnChooseTimeVisitDoctor
-                  key={index}
-                  type='button'
-                  isSelected={isTimeSelect === item}
-                  isDisabled={
-                    item === '2:00 pm' || item === '6:00 pm' ? true : false
-                  }
-                  onClick={(e) => setTimeSelect(e.currentTarget.innerHTML)}
-                >
-                  {item}
-                </StylBtnChooseTimeVisitDoctor>
-              ))}
+              {isPossibleTimeArr.map((item, index) => {
+                const time = new Date(item);
+                const hour = time.getHours();
+                const minutes = time.getMinutes();
+
+                return (
+                  <StylBtnChooseTimeVisitDoctor
+                    key={index}
+                    type='button'
+                    isSelected={isTimeSelect === hour + ':' + minutes + '0'}
+                    onClick={(e) => setTimeSelect(e.currentTarget.innerHTML)}
+                  >
+                    {hour + ':' + minutes + '0'}
```

Comment:
In my opninion it will be clearlier to create component and get rid of this boilerplate code

Reply:
Then you could have only 1 useEffect in component for check button disabled or not. Available time will useEffect in  seperate component. I'm about line 28-32. @codezavr what do you think about it?

Reply:
Fixed
Used moment